src/merge_with.rs

```diff
@@ -0,0 +1,116 @@
+use crate::collection::{CollectionWithId, Id};
+
+pub trait MergeWith<T> {
```

Comment:
This trait is useless if it only used by CollectionWithId
you can directly add your implementation in `collection.rs`

Reply:
I was thinking first of `Collection` too as a possible implementation... and then I realized that inserting in `Collection` wouldn't create conflicts since it's essentially a `Vec`. So I'll move it as a simple method and dump the trait.